R: Fail CI/CD on NOTEs, check examples on UBSAN, log valgrind output

krlmlr opened the pull request:
to avoid discussions on CRAN. To finish, need to revise a7d95a32e. Please advise.

krlmlr committed:
Add job with DUCKDB_R_DEBUG=1

krlmlr committed:
Fail on NOTEs

krlmlr committed:
Avoid NOTEs

krlmlr committed:
Install pandoc

Mytherin commented on src/common/types/vector_constants.cpp:
```diff
@@ -9,8 +9,8 @@ namespace duckdb {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wexit-time-destructors"
+//#pragma clang diagnostic push
+//#pragma clang diagnostic ignored "-Wexit-time-destructors"
```
I think these are required in the OSX builds currently, see [here](https://github.com/duckdb/duckdb/blob/master/src/CMakeLists.txt#L12).

Perhaps we should just remove the warning entirely, or explicitly disable the warning for these files using the CMake instead.

krlmlr replied:
Thanks. One file is ours, the other is vendored. What's the process of updating the vendored version (if we wanted to erase the check there)?

Mytherin replied:
We added the check there, so update in any way you want

Mytherin reviewed (commented):
Thanks for the PR! Looks good to me. One comment:

krlmlr committed:
Always print all logs, separately for regular and valgrind

krlmlr commented:
I'll take a look soon.

Following @hannes's question regarding non-detection of Valgrind errors: we do run the Valgrind checks on CI/CD, but we need also to parse the logs: https://github.com/krlmlr/duckdb/runs/6776149490?check_suite_focus=true#step:8:3046.

I'll take a look at the UBSAN checks too.

james-atkins committed:
R: support structs in queries

This PR allows structs to be returned when running queries from R.
DuckD structs are converted to R data.frames that are nested or packed
inside the primary data frame returned from DuckDB.

As the base R as.data.frame method does not handle nested or packed data
frames well and as DuckDB guarantees that the output will always be a
valid data frame, the necessary attributes (class and number of
rows) are set from C++.

krlmlr committed:
Extract script, run examples

krlmlr committed:
Cleaner output

krlmlr commented:
This now fails the UBSAN check. We don't parse valgrind output yet, but it's logged.

Mytherin commented:
#3812 should fix the valgrind issues that are there but have not been uncovered before

krlmlr committed:
Rerun scripts/generate_flex.py

krlmlr committed:
Don't check incoming

krlmlr committed:
Merge branch 'master' into f-r-debug

krlmlr commented:
As expected: the UBSAN build now fails, consistently with CRAN.

krlmlr commented:
Valgrind log: https://github.com/duckdb/duckdb/runs/6823490797?check_suite_focus=true#step:13:105

hannes committed:
Merge pull request #3495 from james-atkins/r_struct

R: support structs

Mytherin committed:
Merge branch 'f-r-debug' of github.com:krlmlr/duckdb into krlmlr-f-r-debug

Mytherin committed:
Disable exit-time-destructors warning in CMake rather than using pragmas

Mytherin committed:
More warning fixes

Mytherin committed:
More CRAN fixes from Hannes

Mytherin committed:
More CRAN fixes

Mytherin committed:
Indentation fixes

Mytherin committed:
library(arrow)

Mytherin committed:
Like this maybe

Mytherin committed:
Remove more regex

krlmlr commented:
Pulled in the changes from https://github.com/Mytherin/duckdb/tree/krlmlr-f-r-debug. Builds were green there.

Mytherin commented:
Thanks!
